core/templates/components/state-directives/outcome-editor/outcome-destination-editor.component.html

```diff
@@ -28,14 +28,17 @@
   </form>
   <p [hidden]="newStateNameForm.form.valid ||  outcomeNewStateName === undefined" class="form-text oppia-form-error oppia-form-error-text">Please pick a card name consisting of alphanumeric characters, spaces and/or hyphens.</p>
 
-  <div *ngIf="isSelfLoop() && canEditRefresherExplorationId" class="oppia-id-text">
+  <div *ngIf="isSelfLoop() && canEditRefresherExplorationId && !isrefresherExplorationIdNull(outcome.refresherExplorationId)" class="oppia-id-text">
```

Comment:
Perhaps just `!!outcome.refresherExplorationId` will be sufficient. Note that `!!null` and `!!''` will evaluate to false.

Reply:
Oh!
Got it, Thanks.